File: rust/binaryen-decompile/tests/_loop_sum.roundtrip.cpp
// Decompiled from WebAssembly

int32_t func_0(int32_t idx) {
  int32_t v = 0;
  {
    {
      if ((idx >= 1)) { goto label$0; };
      return 0;
    }
  }
label$0: ;
  return ((v = (idx + -1)) + (i32)(((i64)(u32)v * (i64)(u32)(idx + -2)) >>> 1));
}

File: rust/binaryen-decompile/tests/_loop_sum.roundtrip.rs
// Decompiled to Rust from WebAssembly

fn func_0(idx: i32) -> i32 {
    let mut v: i32 = 0;
    'label$0: {
        {
            if (idx >= 1) { break 'label$0 };
            return 0;
        }
    }
    ({ v = (idx + -1); } + ((((v as u64) * ((idx + -2) as u64)) >> 1i64) as i32))
}
